### DISCUSSION TITLE
How to open a single chrome profile directly with seleniumbase?

### DISCUSSION THREAD

**Discussion by pd2871** [seleniumbase/SeleniumBase#2118] [Q&A] [answered]
Hi @mdmintz 
I am trying to open a Chrome profile using seleniumbase but am not able to find any docs related to it. Please help me on it.

Currently, I have tried the below method but it shows the list of profiles instead of opening a single chrome profile.

```
with SB(uc=True, user_data_dir=r'C:\Users\Acer\AppData\Local\Google\Chrome\User Data') as sb: 
    sb.driver.get('https://google.com')
```

**Comment by mdmintz** [comment, marked as answer]
If using the `user_data_dir` option together with `uc` mode, the `user_data_dir` MUST be one that was created via tests running in `uc` mode. This means that you cannot use one that comes from your default Chrome browser. A regular Chrome browser would reconfigure the `user_data_dir` in a way such that `uc` mode would no longer be compatible with it.

Also, if using Windows, make sure that the path to your user_data_dir (and the user_data_dir itself) does not contain any spaces. There appears to be an external issue on Windows (I've heard) with selenium, undetected-chromedriver, and pyinstaller (outside the scope of SeleniumBase) which may cause problems if that path has spaces in it.

**Reply by caffenie** [reply]
how can the browser profile created during uc mode be stored?

**Reply by mdmintz** [reply]
Unless manually deleted, any custom `user_data_dir` created will still be there after the UC Mode test completes. Then be sure to specify the same `user_data_dir` again.

(Note that you can configure the `user_data_dir`, but not the Profile directory inside that, which is separate. Only the `Default` profile is used, and that is automatically selected when specifying the `user_data_dir`.)

**Reply by caffenie** [reply]
where can I find the custom user_data_dir created in UC mode? I cannot find any functions for it, and specfiying the user_data_dir param to an empty folder makes selenium unable to launch. 

Also, thank you for all your hard work

**Reply by mdmintz** [reply]
Don't specify an empty folder when setting the `user_data_dir`. Either use a folder that doesn't exist, or an existing folder that was created previously when you set a `user_data_dir` to use.

**Reply by floresernesto95** [reply]
> Don't specify an empty folder when setting the `user_data_dir`. Either use a folder that doesn't exist, or an existing folder that was created previously when you set a `user_data_dir` to use.

Thanks man for taking the time to answer, also help me!
